auem/train.py

```diff
@@ -28,23 +42,49 @@ def train(cfg: DictConfig) -> None:
 
     criterion = hydra.utils.instantiate(cfg.criterion)
 
+    num_batches_train = ceil(len(ds_train)/cfg.dataloader.params.batch_size)
+    num_batches_valid = ceil(len(ds_valid)/cfg.dataloader.params.batch_size)
+
+    writer = tb.SummaryWriter()
+    writer.add_graph(model, iter(dl_train).next()['X'].to(device))
     for epoch in tqdm(range(cfg.epochs), position=0, desc="Epoch"):
-        for batch in tqdm(dataloader, total=ceil(len(dataset)/cfg.dataloader.params.batch_size), position=1, desc="Batch"):
+        losses, items_seen = 0, 0
+        model.train()
+        for batch_num, batch in tqdm(enumerate(dl_train), total=num_batches_train, position=1, desc="Batch"):
+            gs += 1
             X, y = batch["sample"].to(device), batch["label"].to(device)
             optimizer.zero_grad()
             output = model(X)
             loss = criterion(output, y)
+            losses += loss.item()
+            items_seen += batch.shape[0]
             loss.backward()
             optimizer.step()
-
-    logger.debug(f"in train loop {cfg.intrain}")
-
+        writer.add_scalar(f"loss/training", losses/batch.shape[0], global_step=epoch)
+        # writer.add_scalars(f"accuracy/training", accuracies, global_step=epoch)
+        # evaluation loop
+        if cfg.eval:
```

Comment:
as i said before, in a future MR it might be nice to modularize this a bit more, but not necessary for this one